src/metrics/src/lgalloc.rs

```diff
@@ -93,22 +91,25 @@ macro_rules! metrics_size_class {
                     })
                 }
                 fn update(&mut self) {
-                    let mut stats = std::mem::take(&mut self.stats);
-                    lgalloc::lgalloc_stats(&mut stats);
+                    let stats = lgalloc::lgalloc_stats();
                     for sc in &stats.size_class {
                         let sc_stats = self.get_size_class(sc.size_class);
                         $(sc_stats.$metric.set(($conv)(u64::cast_from(sc.$name), sc));)*
                     }
                     let mut accums = BTreeMap::new();
-                    for file_stat in &stats.file_stats {
-                        let accum: &mut FileStatsAccum = accums.entry(file_stat.size_class).or_default();
-                        accum.add_assign(file_stat);
+                    match &stats.file_stats {
+                        Ok(file_stats) => {
+                            for file_stat in file_stats {
+                                let accum: &mut FileStatsAccum = accums.entry(file_stat.size_class).or_default();
+                                accum.add_assign(file_stat);
+                            }
+                        }
+                        Err(err) => eprintln!("Failed to get file stats: {err}"),
```

Comment:
We might want to return the error from Update so that on error we stop the task that collects metrics.